examples/todomvc/src/element.rs

```diff
@@ -0,0 +1,235 @@
+                                let dyn_target_el: Option<
+                                    &web_sys::Node,
+                                > = wasm_bindgen::JsCast::dyn_ref(&target_element);
+                                if let Some(target_element) = dyn_target_el {
+                                    if let Ok(potential_elements) =
+                                        tg_el.query_selector_all(selector)
+                                    {
+                                        let mut has_match = false;
+                                        /*
+                                        dbg(format!(
+                                            "len: {} {}",
+                                            potential_elements.length(),
+                                            selector
+                                        ).as_str());
+*/
+                                        for i in 0..potential_elements.length() {
+                                            if let Some(el) = potential_elements.get(i) {
+                                                if target_element.is_equal_node(Some(&el)) {
+                                                    has_match = true;
+                                                }
+                                                break;
+                                            }
+                                        }
+
+                                        if has_match {
+                                            handler(event);
+                                        }
+                                    }
+                                }
+                            }
+                        }) as Box<FnMut(_)>);
+
+                        dyn_el.add_event_listener_with_callback_and_bool(
+                            //element_et.add_event_listener_with_callback_and_bool(
+                            event,
+                            cb.as_ref().unchecked_ref(),
+                            use_capture,
+                        );
+                        cb.forget(); // TODO cycle collect
+                    }
+                }
+            }
+            self.el = Some(el);
+        }
+    }
+
+    /// Find child `Element`s from this node
+    pub fn qs_from(&mut self, selector: &str) -> Option<Element> {
+        let mut found_el = None;
+        if let Some(el) = self.el.take() {
+            found_el = Some(Element {
+                el: el.query_selector(selector).ok()?,
+            });
+            self.el = Some(el);
+        }
+        found_el
+    }
+
+    /// Sets the inner HTML of the `self.el` element
+    pub fn set_inner_html(&mut self, value: String) {
+        if let Some(el) = self.el.take() {
+            el.set_inner_html(&value);
+            self.el = Some(el);
+        }
+    }
+
+    /// Sets the text content of the `self.el` element
+    pub fn set_text_content(&mut self, value: String) {
+        if let Some(el) = self.el.take() {
```

Comment:
This pattern seems a little odd to me, is there a reason `.as_ref()` without moving ownership wasn't used?

Reply:
Mostly because I forgot `as_ref` was possible :laughing: I will simplify where possible :) thanks! (and for all the other reviews).

Reply:
Oh I think mostly because `.into` needs ownership so I got into the habit of needing ownership.